subprojects/language-native/src/main/java/org/gradle/language/swift/internal/DefaultSwiftBinary.java

```diff
@@ -150,4 +174,73 @@ public FileCollection getObjects() {
     public Property<SwiftCompile> getCompileTask() {
         return compileTaskProperty;
     }
+
+    @Inject
+    protected NativeDependencyCache getNativeDependencyCache() {
+        throw new UnsupportedOperationException();
+    }
+
+    private class ModulePath implements MinimalFileSet, Buildable {
+        private final Configuration importPathConfig;
+
+        private Set<File> result;
+
+        ModulePath(Configuration importPathConfig) {
+            this.importPathConfig = importPathConfig;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Module include path for " + DefaultSwiftBinary.this.toString();
+        }
+
+        @Override
+        public Set<File> getFiles() {
+            if (result == null) {
+                result = Sets.newLinkedHashSet();
+                Map<String, ModuleMap> moduleMaps = Maps.newHashMap();
+                for (ResolvedArtifactResult artifact : importPathConfig.getIncoming().getArtifacts()) {
+                    Usage usage = artifact.getVariant().getAttributes().getAttribute(Usage.USAGE_ATTRIBUTE);
+                    if (usage != null && Usage.C_PLUS_PLUS_API.equals(usage.getName())) {
+                        String moduleName;
+
+                        ComponentIdentifier id = artifact.getId().getComponentIdentifier();
```

Comment:
In the long run, would the module name be another attribute of the variant?

I think this makes the assumption that artifacts with a Usage == C_PLUS_PLUS_API name their modules the same as their project names or publication names.

Reply:
It could be, I suppose, but again that presumes that the producer should have some say in how the consumer refers to him.  I mean, only the consumer knows what module he is importing and I'm not sure that module name has to be universal.  I was expecting any sort of configurability here to be on the consumer side (i.e. I use the "foo" library, but in my build, I refer to it as "bar" for some reason known only to me).  This (how we enable configurability) definitely deserves some discussion when we get to that point.